examples/model_loader.py

```diff
@@ -0,0 +1,94 @@
+from yggdrasil.examples import yamls as ex_yamls
+import yaml
+import jupyterlab_nodeeditor as jlne
+
+def dict_conversion(model_file):
+    # Use flags to determine model input type
+    is_dict, is_list = False, False
+    
+    # There is no clear indicator if a safeloaded YAML will have single or multiple models
+    # List parsing is incomplete at the moment, but all models tested were dictionaries
+    
+    # If it's neither, return the type so it can be noted and accounted for later
+    if type(model_file) is dict:
+        is_dict = True
+    elif type(model_file) is list:
+        is_list = True
+    else:
+        return ("Unaccounted model type: ", type(model_file))
+    
+    # This works with both single and multiple model dictionaries
+    # If the model type is a list, please notify Het
+    if is_dict:       
+        list_of_models = []
+        
+        if "models" in model_file.keys():
+            for model in model_file["models"]:
+                # Setup dictionary to be filled
+                new_dict, new_dict["inputs"], new_dict["outputs"], new_dict['title'] = {}, [], [], model['name']
+                
+                # Fill in the Inputs
+                if "input" in model.keys():
+                    for i, inp in enumerate(model["input"]):
+                        new_dict["inputs"].append({'title': inp["name"], 'key': f"temp_in{i}", 'socket_type': "connection"})
+                elif "inputs" in model.keys():
+                    for i, inp in enumerate(model["inputs"]):
+                        new_dict["inputs"].append({'title': inp["name"], 'key': f"temp_in{i}", 'socket_type': "connection"})
+        
+                # Fill in the Outputs, same as inputs with name changes
+                if "output" in model.keys():
+                    for o, out in enumerate(model["output"]):
+                        new_dict["outputs"].append({'title': out["name"], 'key': f"temp_out{o}", 'socket_type': "connection"})
+                elif "outputs" in model.keys():
+                    for o, out in enumerate(model["outputs"]):
+                        new_dict["outputs"].append({'title': out["name"], 'key': f"temp_out{o}", 'socket_type': "connection"})
+                list_of_models.append(new_dict)
+        
+        # We return a list here so that way loading the models is uniform for both multi-model and single-model files
+        return list_of_models
+
+
+# ps - Node Editor instance that it is added to, default blank
+def load_model(ps = None):
```

Comment:
Is this a public-facing function?  If so, it needs a docstring.

Reply:
Added a docstring since it is public-facing